--- CA-org.pl	2020-03-17 22:50:21.000000000 +0900
+++ CA.pl	2020-03-17 22:50:33.000000000 +0900
@@ -26,7 +26,7 @@ my $verbose = 1;
 
 my $OPENSSL_CONFIG = $ENV{"OPENSSL_CONFIG"} || "";
 my $DAYS = "-days 365";
-my $CADAYS = "-days 1095";	# 3 years
+my $CADAYS = "-days 3650";	# 10 years
 my $REQ = "$openssl req $OPENSSL_CONFIG";
 my $CA = "$openssl ca $OPENSSL_CONFIG";
 my $VERIFY = "$openssl verify";
@@ -34,7 +34,7 @@ my $X509 = "$openssl x509";
 my $PKCS12 = "$openssl pkcs12";
 
 # default openssl.cnf file has setup as per the following
-my $CATOP = "./demoCA";
+my $CATOP = "/home/CA";
 my $CAKEY = "cakey.pem";
 my $CAREQ = "careq.pem";
 my $CACERT = "cacert.pem";
